packages/flame_tiled/lib/src/tiled_component.dart

```diff
@@ -79,49 +95,62 @@ class TiledComponent<T extends FlameGame> extends PositionComponent
     );
   }
 
-  static Vector2 _computeSize(RenderableTiledMap tileMap) {
-    final tMap = tileMap.map;
-
-    final xScale = tileMap.destTileSize.x / tMap.tileWidth;
-    final yScale = tileMap.destTileSize.y / tMap.tileHeight;
+  @visibleForTesting
+  static Vector2 computeSize(
+    MapOrientation? orientation,
+    Vector2 destTileSize,
+    int tileWidth,
+    int tileHeight,
+    int mapWidth,
+    int mapHeight,
+    StaggerAxis? staggerAxis,
+  ) {
+    if (orientation == null) {
+      return NotifyingVector2.zero();
+    }
+    final xScale = destTileSize.x / tileWidth;
+    final yScale = destTileSize.y / tileHeight;
 
     final tileScaled = Vector2(
-      tileMap.map.tileWidth * xScale,
-      tileMap.map.tileHeight * yScale,
+      tileWidth * xScale,
+      tileHeight * yScale,
     );
 
-    if (tMap.orientation == null) {
-      return NotifyingVector2.zero();
-    }
-
-    switch (tMap.orientation!) {
+    switch (orientation) {
       case MapOrientation.staggered:
-        return tMap.staggerAxis == StaggerAxis.y
+        return staggerAxis == StaggerAxis.y
             ? Vector2(
-                tileScaled.x * tileMap.map.width + tileScaled.x / 2,
-                tileScaled.y + ((tileMap.map.height - 1) * tileScaled.y / 2),
+                tileScaled.x * mapWidth + tileScaled.x / 2,
+                (mapHeight + 1) * (tileScaled.y / 2),
               )
             : Vector2(
-                tileScaled.x + ((tileMap.map.width - 1) * tileScaled.x / 2),
-                tileScaled.y * tileMap.map.height + tileScaled.y / 2,
+                (mapWidth + 1) * (tileScaled.x / 2),
+                tileScaled.y * mapHeight + tileScaled.y / 2,
               );
 
       case MapOrientation.hexagonal:
-        return tMap.staggerAxis == StaggerAxis.y
+        return staggerAxis == StaggerAxis.y
             ? Vector2(
-                tileMap.map.width * tileScaled.x + tileScaled.x / 2,
-                tileScaled.y + ((tileMap.map.height - 1) * tileScaled.y * 0.75),
+                mapWidth * tileScaled.x + tileScaled.x / 2,
+                tileScaled.y + ((mapHeight - 1) * tileScaled.y * 0.75),
               )
             : Vector2(
-                tileScaled.x + ((tileMap.map.width - 1) * tileScaled.x * 0.75),
-                (tileMap.map.height * tileScaled.y) + tileScaled.y / 2,
+                tileScaled.x + ((mapWidth - 1) * tileScaled.x * 0.75),
+                (mapHeight * tileScaled.y) + tileScaled.y / 2,
               );
 
       case MapOrientation.isometric:
+        final dimensions = mapWidth + mapHeight;
+        final halfTile = tileScaled / 2;
+        return Vector2(
+          halfTile.x * dimensions,
+          halfTile.y * dimensions,
+        );
```

Comment:
```suggestion
        final halfTile = tileScaled / 2;
        return halfTile..scale(dimensions);
```
Creates one less `Vector2` object.

Reply:
👍🏻